internal/infrastructure/kubernetes/service.go

```diff
@@ -0,0 +1,118 @@
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/envoyproxy/gateway/internal/ir"
+)
+
+const (
+	// envoyServiceHTTPPort is the HTTP port number of the Envoy service.
+	envoyServiceHTTPPort = int32(80)
+	// envoyServiceHTTPSPort is the HTTPS port number of the Envoy service.
+	envoyServiceHTTPSPort = int32(443)
```

Comment:
Any reason to have these typed as `int32` rather than just letting them be [untyped constants](https://go.dev/blog/constants)?

(I prickle every time I see an port number typed as something other than the proper `uint16`; Kubernetes' excuse is that the Kube APIs were originally canonically specified as protobuf, and protobuf doesn't have 16-bit number types.)